channels/pkg/channels/apply.go

```diff
@@ -50,6 +50,11 @@ func Apply(data []byte) error {
 		if err != nil {
 			klog.Errorf("failed to apply the manifest: %v", err)
 		}
+		/*
+			if err == nil {
+				return nil
+			}
+		*/
```

Comment:
Any reason for this comment to be added?
/lgtm

Reply:
Yeah ... I was contemplating if we should do this or not.

Plus side is that more addons will be considered updated for lower k8s versions.
Downside is that in some cases, e.g containers we have removed from the manifest won't be won't be removed from actual.

Maybe apply, replace, apply and only check the error of the last apply is the way to go.

Reply:
> Maybe apply, replace, apply and only check the error of the last apply is the way to go.

Sounds reasonable to me.